### DISCUSSION TITLE
KV/Bucket predefined configuration

### DISCUSSION THREAD

**Discussion by paulorb** [nats-io/nats-server#5591] [Q&A]
Hi everyone,

  Is there any way configure Nats server by command line argument or configuration file to create a static, pre-defined KV/Bucket for JetStream.  Currently I have a script that after nats is running it invokes nats cli and create the bucket/kv. I would prefer to start nats server with this configuration already loaded. Is it possible? Where is it documented?

nats kv add config
nats kv put config factype default

Thanks,
Paulo

**Comment by neilalexander** [comment]
There's no built-in way to do this, no.

Depending on your environment, you might consider using one of the [Configuration Management](https://docs.nats.io/running-a-nats-service/configuration/resource_management/configuration_mgmt) tools for this purpose.
